lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,208 @@
+            completedFactors,
+            userContext: ctx,
+        });
+        await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.assertAllowedToSetupFactorElseThrowInvalidClaimError(
+            {
+                session,
+                factorId,
+                completedFactors,
+                mfaRequirementsForAuth: requirements,
+                factorsSetUpForUser: factorsSetup,
+                requiredSecondaryFactorsForUser: defaultMFARequirementsForUser,
+                requiredSecondaryFactorsForTenant: defaultMFARequirementsForTenant,
+                userContext: ctx,
+            }
+        );
+    }
+
+    static async getMFARequirementsForAuth(session: SessionContainerInterface, userContext?: Record<string, any>) {
+        let ctx = getUserContext(userContext);
+        const user = await getUser(session.getUserId(), ctx);
+        if (!user) {
+            throw new Error("Session user not found");
+        }
+        const factorsSetup = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            tenantId: session.getTenantId(),
+            user,
+            userContext: ctx,
+        });
+        const mfaClaimValue = await session.getClaimValue(MultiFactorAuthClaim, ctx);
+        const completedFactors = mfaClaimValue?.c ?? {};
+        const defaultMFARequirementsForUser = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getRequiredSecondaryFactorsForUser(
+            {
+                userId: session.getUserId(),
+                userContext: ctx,
+            }
+        );
+
+        const tenantInfo = await Multitenancy.getTenant(session.getTenantId(), userContext);
+
+        if (tenantInfo === undefined) {
+            throw new Error("Tenant not found");
+        }
+
+        const defaultMFARequirementsForTenant: string[] = tenantInfo.requiredSecondaryFactors ?? [];
+        return await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getMFARequirementsForAuth({
+            user,
+            accessTokenPayload: session.getAccessTokenPayload(),
+            tenantId: session.getTenantId(),
+            factorsSetUpForUser: factorsSetup,
+            requiredSecondaryFactorsForUser: defaultMFARequirementsForUser,
+            requiredSecondaryFactorsForTenant: defaultMFARequirementsForTenant,
+            completedFactors,
+            userContext: ctx,
+        });
+    }
+
+    static async markFactorAsCompleteInSession(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.markFactorAsCompleteInSession({
+            session,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async getFactorsSetupForUser(tenantId: string, userId: string, userContext?: Record<string, any>) {
```

Comment:
this function should no longer require a tenantId as an input

Reply:
Done